src/message.js

```diff
@@ -126,56 +124,64 @@ Message.prototype.decrypt = function(privateKey, sessionKey, password) {
  *                               { data:Uint8Array, algorithm:String }
  */
 Message.prototype.decryptSessionKey = function(privateKey, password) {
-  var keyPacket;
-
-  if (password) {
-    var symEncryptedSessionKeyPacketlist = this.packets.filterByTag(enums.packet.symEncryptedSessionKey);
-    var symLength = symEncryptedSessionKeyPacketlist.length;
-    for (var i = 0; i < symLength; i++) {
-      keyPacket = symEncryptedSessionKeyPacketlist[i];
-      try {
-        keyPacket.decrypt(password);
-        break;
+  var keyPacket, results, error;
+  return Promise.resolve().then(async () => {
+    if (password) {
+      var symESKeyPacketlist = this.packets.filterByTag(enums.packet.symEncryptedSessionKey);
+      if (!symESKeyPacketlist) {
+        throw new Error('No symmetrically encrypted session key packet found.');
       }
-      catch(err) {
-        if (i === (symLength - 1)) {
-          throw err;
+      // FIXME need a circuit breaker here
+      results = await Promise.all(symESKeyPacketlist.map(async function(packet) {
```

Comment:
I think the "circuit breaker" that you want can just be achieved by using a for loop here instead of a promise all. In general I think this can be applied to other parts of your code (sacrificing the parallelization for fast-failing and code simplicity).

Reply:
ESLint would not allow this: https://eslint.org/docs/rules/no-await-in-loop. We can silence this, of course, but I do think it's worth it overall and it doesn't make the code that much more complicated. Also, I believe a "circuit breaker" for promises is on the way.

Reply:
I had a chat with Bart and some of our Javascript frontend guys, and we came to the conclusion that a for loop would be cleaner and actually more of a standard in these cases (when you would break out of the loop after success/failures). We can just add /* eslint no-await-in-loop: "off" */ here to silence these particular cases rather than changing our eslint config altogether.